runtime/vm/jvminit.c

```diff
@@ -2124,8 +2136,14 @@ VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved)
 #endif /* defined(J9VM_ARCH_X86) || defined(J9VM_ARCH_S390) || defined(J9VM_ARCH_POWER) || defined(J9VM_ARCH_AARCH64) */
 					}
 				}
-
-
+#if (defined(J9VM_ARCH_X86) || defined(J9VM_ARCH_S390) || defined(J9VM_ARCH_POWER) || defined(J9VM_ARCH_AARCH64)) && defined(J9VM_OPT_JITSERVER)
+				// Regardless of whether or not the local SCC is enabled, we still need to enable portable AOT if
```

Comment:
I think this change was @mpirvu's suggestion. It was in the context of a discussion we were having about the `TR_AOTHeader` differing depending on whether or not `-Xshareclasses` was enabled or not.

Reply:
This needs to be fixed eventually, it's a performance degradation. I remember seeing measurable reductions in throughput with portable AOT back when I worked on the original AOT cache implementation. There is also no point in enabling portable AOT by default at all when running in a container without local SCC (but with AOT cache). Could you please open an issue for this?

I still don't understand what could be different in the `TR_AOTHeader` depending on whether local SCC is enabled. Maybe there is something subtle going wrong with either `featureFlags` or `processDescription`. Have you tried testing this both inside and outside a container, and with portable AOT explicitly enabled or disabled?

Reply:
Initially I had individual overrides in particular places, because having/not having an SCC would change some VM properties at startup. Those solved all the header incompatibilities. Explicitly setting the relocatable processor/CPU description in `rossa.cpp` was one of them. Changing how the compressed pointer shift amount was determined was another, as that changes if we're in containers and using an SCC (from what I remember).

Setting the portable AOT flag happened to fix all those individual issues.

Reply:
I'll open up an issue